lib/appsignal/hooks/active_support_notifications.rb

```diff
@@ -0,0 +1,35 @@
+module Appsignal
+  class Hooks
+    class ActiveSupportNotificationsHook < Appsignal::Hooks::Hook
+      register :active_support_notifications
+
+      def dependencies_present?
+        defined?(::ActiveSupport::Notifications)
+      end
+
+      def install
+        ::ActiveSupport::Notifications.class_eval do
+          class << self
+            alias instrument_without_appsignal instrument
+
+            def instrument(name, payload={}, &block)
+              transaction = Appsignal::Transaction.current
+
+              transaction.start_event
+
+              instrument_without_appsignal(name, payload, &block)
+
+              title, body, body_format = Appsignal::EventFormatter.format(name, payload)
+              transaction.finish_event(
+                name,
+                title,
+                body,
+                body_format
+              )
```

Comment:
We could use the instrument helper, but that would just add another layer of a block being called and would have no real benefit.

These method calls will do nothing when active is false. Also in that case this hook would not be installed at all.